### ISSUE TITLE
injectedJavaScript isn't properly escaped on Android

### ISSUE THREAD

**Issue by LinusU** [facebook/react-native#20365]
## Environment
```
  React Native Environment Info:
    System:
      OS: macOS High Sierra 10.13.5
      CPU: x64 Intel(R) Core(TM) i7-5557U CPU @ 3.10GHz
      Memory: 996.13 MB / 16.00 GB
      Shell: 5.5.1 - /usr/local/bin/zsh
    Binaries:
      Node: 10.5.0 - /usr/local/bin/node
      Yarn: 1.7.0 - /usr/local/bin/yarn
      npm: 6.1.0 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 11.4, macOS 10.13, tvOS 11.4, watchOS 4.3
      Android SDK:
        Build Tools: 23.0.1, 26.0.3, 27.0.3
        API Levels: 19, 23, 26
    IDEs:
      Xcode: 9.4.1/9F2000 - /usr/bin/xcodebuild
    npmPackages:
      @fanny-pack/react-native: ^1.0.0-beta.1 => 1.0.0-beta.1 
      @types/react: ^16.4.6 => 16.4.6 
      @types/react-native: ^0.55.26 => 0.55.28 
      react: 16.4.1 => 16.4.1 
      react-native: ^0.56.0 => 0.56.0 
    npmGlobalPackages:
      create-react-native-app: 1.0.0
      react-native-cli: 2.0.1
```

## Description
Any JavaScript passed as the `injectedJavaScript` property on the `WebView` component on Android isn't properly escaped, and thus can lead to strange SyntaxErrors.

This can be seen here as the raw string is being passed as an `javascript:`-uri:

https://github.com/facebook/react-native/blob/6e359c4589b0627de59332ce56fe28804425a609/ReactAndroid/src/main/java/com/facebook/react/views/webview/ReactWebViewManager.java#L328-L334

## Reproducible Demo
https://snack.expo.io/rkSxWhNVm

On iOS, three "12" are correctly showing up. On Android, only the first "12" is printed.

**Comment by eranbaroz** [comment]
I agree that this solution is "shitty hack", but if you need to convert to base64  
than a better simple way to encode to base64 can be found in react-native-base64 npm package  
[https://www.npmjs.com/package/react-native-base64](https://www.npmjs.com/package/react-native-base64)

**Comment by LinusU** [comment]
@eranbaroz I don't think that that is a better solution. That package is only at `0.0.1`, and only has 50 downloads per week. It doesn't use anything that's special to the React Native platform either so I don't see why it would be a better approach than the packages I linked.

But the most compelling reason not to use it is that it doesn't work. I actually used that package first, but `window.atob` failed to decode the result.

**Comment by LinusU** [comment]
Does anyone know why @react-native-bot marks all my issues as "Old Version", even though I'm running 0.56 which should be the latest? (ping @kelset ☺️)

**Comment by eranbo** [comment]
@LinusU in some cases it might be right. But this  specific package is only 1 week old or less and published by yours truly (me:)).  
Off-course it doesn't use anything special to the RN platform, this is why this package can be used in any other platform/framework that is javascript based and not running in a browser with window.btoa() / window.atob().  
The package you linked do require us to link the package to the project and modifies native code, and probably won't work in create-react-native-app application that runs inside expo, only on ejected applications.

**Comment by LinusU** [comment]
> But this specific package is only 1 week old or less and published by yours truly (me:))

Unfortunately, I don't know who you are so I can't attest whether or not the package is stable. That being said, I fail to see how it would be "better" than my approach, please explain ☺️ 

> The package you linked do require us to link the package to the project and modifies native code, and probably won't work in create-react-native-app application that runs inside expo, only on ejected applications.

That's incorrect, the linked package is simply a small, pure javascript dependency. You can see the sources here:

- https://github.com/beatgammit/base64-js/blob/master/index.js
- https://github.com/LinusU/encode-utf8/blob/master/index.js

But as I said, your package didn't work, so I would not recommend using that package until at least that is fixed.
